File: sources/scalac/transformer/TransMatchPhase.java
/*     ____ ____  ____ ____  ______                                     *\
**    / __// __ \/ __// __ \/ ____/    SOcos COmpiles Scala             **
**  __\_ \/ /_/ / /__/ /_/ /\_ \       (c) 2002, LAMP/EPFL              **
** /_____/\____/\___/\____/____/                                        **
**                                                                      **
** $Id$
\*                                                                      */

package scalac.transformer;

import scalac.*;
import scalac.checkers.*;

public class TransMatchPhase extends PhaseDescriptor {

    public String name () {
        return "transmatch";
    }

    public String description () {
        return "translate match expressions";
    }

    public String taskDescription() {
        return "translated pattern matching";
    }

    public Phase createPhase(Global global) {
        return new TransMatch(global, this);
    }

    public Checker[] postCheckers(Global global) {
        return new Checker[] {
            new CheckSymbols(global),
            new CheckTypes(global),
            new CheckOwners(global),
	    new CheckNames(global)
        };
    }
}

File: sources/scala/tools/scalac/transformer/TransMatchPhase.scala
/*     ____ ____  ____ ____  ______                                     *\
**    / __// __ \/ __// __ \/ ____/    SOcos COmpiles Scala             **
**  __\_ \/ /_/ / /__/ /_/ /\_ \       (c) 2002, LAMP/EPFL              **
** /_____/\____/\___/\____/____/                                        **
\*                                                                      */

// $Id$

import scalac.{Global => scalac_Global}
import scalac.Phase;
import scalac.PhaseDescriptor;
import scalac.CompilationUnit;
import scalac.checkers._;

package scala.tools.scalac.transformer {

class TransMatchPhase(global:scalac_Global, descriptor:PhaseDescriptor )
  extends Phase(global, descriptor) {

    /** Applies this phase to the given compilation units. */
    def apply( units:Array[CompilationUnit] ):unit = {
      for(val u <- units) {
        new TransMatch( global ).apply( u );
      }
    }
    override def postCheckers( global:scalac_Global ):Array[Checker] = {
      val a = new Array[Checker](2);
      a.update(0, new CheckSymbols(global) );
      a.update(1, new CheckTypes(global) );
      a
    }
  }
}


